### PR TITLE
Testgrid Configuration Sharding

### PR DESCRIPTION
**Pull request by chases2**
Testgrid configurations have been moved to an external directory, for two reasons:
- Configurations can now be "sharded" among any number of YAML config files, instead of two monolithic files
- Configuration information specific to our _instance_ of Testgrid is no longer in /testgrid

A requirement was added that each YAML file have a "default" definition to curtail undefined behavior. The behavior was already there and undocumented; the order that config files were specified in used to determine default behavior. By requiring each file to have a "default" definition, each file can specify its own defaults and the defaulting behavior is more obvious

/cc @michelle192837 
/cc @Katharine

### CODE REVIEW

File: testgrid/cmd/configurator/main.go [hunk 1]
```diff
@@ -144,18 +160,67 @@ func announceProwChanges(ctx context.Context, pca *prowConfig.Agent, channel cha
...
+		}
+
+		err = filepath.Walk(path, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				logrus.WithError(err).Errorf("walking path %q.", path)
+				// bad file should not stop us from parsing the directory
+				return nil
+			}
+
+			if strings.HasPrefix(info.Name(), "..") {
```
Comment (Katharine) [inline testgrid/cmd/configurator/main.go]: I'm curious what files are being added with a `..` prefix?

Comment (chases2) [reply testgrid/cmd/configurator/main.go]: I took a number of hints from the way that Prow does this to determine the best way to traverse and monitor these files

When Prow is digging through jobs, it ignores ".." directories in this way. [Example](https://github.com/kubernetes/test-infra/blob/a3c842850be5c9c8681d38ce52c03163e2d918ca/prow/config/config.go#L504-L506)
Honestly, I can't tell you exactly why; filepath.Walk ignores the obvious ".." link to go up a directory. From the comment, it seems that kubernetes volumes contain actual files prefixed by ".." that need to be explicitly ignored

@krzyzacy , do you know why this code is there in Prow? [Commit](https://github.com/kubernetes/test-infra/commit/41f7bc25f51bce38b3cc89f3bfac2544789c0c34)

Comment (krzyzacy) [reply testgrid/cmd/configurator/main.go]: ref https://github.com/kubernetes/test-infra/pull/8454#issuecomment-400162513

It's a Kubernetes storage specific thing (it auto creates subdir with `..****` in a configmap), and probably has nothing to do with testgrid here.

**Comment by chases2** [comment]
Added preamble to generate_tests.py and deleted the ".." check

**Comment by spiffxp** [comment]
ref https://github.com/kubernetes/test-infra/issues/13541

**Pull request merged**